cibuildwheel/macos.py

```diff
@@ -49,37 +51,65 @@ def call(args, env=None, cwd=None, shell=False):
             print('+ ' + ' '.join(shlex_quote(a) for a in args))
 
         return subprocess.check_call(args, env=env, cwd=cwd, shell=shell)
-
-    # get latest pip once and for all
-    download(get_pip_url, get_pip_script)
-
-    for config in python_configurations:
-        # if this version of python isn't installed, get it from python.org and install
-        python_package_identifier = 'org.python.Python.PythonFramework-%s' % config.version
+  
+    def install_cpython(version, url):
```

Comment:
why this functions are in `build` function body? I think that it reduces code readability and I prefer to change it in top level function.

Reply:
I did this because `shell` and `simple_shell` did the same. Just looked it up, that's older than 3 years. @joerick, do you remember why you chose to nest `shell`?

But yeah, I could put it outside, I guess; though it does not really bother me that it's nested.